FrontEnd/src/components/Filters/GenderFilter/GenderFilter.tsx

```diff
@@ -1,29 +1,94 @@
-import { useDispatch } from "react-redux"
+import { useDispatch, useSelector } from "react-redux"
 import CustomCheckBox from "../../CustomCheckBox/CustomCheckBox"
 import styles from "./GenderFilter.module.css"
-import { updateGender } from "../../../store/Features/FiltersSlice/FiltersSlice"
+import {
+  resetGender,
+  updateGender,
+} from "../../../store/Features/FiltersSlice/FiltersSlice"
+import { useLocation, useNavigate } from "react-router-dom"
+import { useEffect } from "react"
+import { initialStateType as FiltersStateType } from "../../../store/Features/FiltersSlice/FiltersSlice"
+
+interface ReduxState {
+  filters: FiltersStateType
+}
 
 function GenderFilter() {
   const dispatch = useDispatch()
+  const { states } = useSelector((state: ReduxState) => state.filters)
```

Comment:
no need to use entire filters state, it will cause unnecessary re-renders, if you need only genders state just select only them.